hw/ip/i2c/data/i2c_testplan.hjson

```diff
@@ -608,29 +613,83 @@
+              - If the 'controller_halt' interrupt is unhandled, after the HOST_NACK_HANDLER_TIMEOUT delay
+                the Controller FSM should be observed returning to Idle, and a Stop condition should have
+                been generated to end the active transaction.
+              - Check that the CONTROLLER_EVENTS.UNHANDLED_NACK_TIMEOUT bit is set when the timeout is
+                reached.
+            '''
+      stage: V2
+      tests: []
+    }
+    {
+      name: target_mode_n_byte_ack_control
+      desc: '''
+            Test the (Target-Mode) N-Byte ACK control mode
+            When this mode is enabled, we should expect:
+            - So long as there is space in the ACQ_FIFO, write data bytes will be ACK'd if the
+              auto-decremeting ACK Counter (TARGET_ACK_CTRL.NBYTES) contains a positive value.
+            - If the Auto ACK Counter is zero, the clock is stretched up to the timeout configured in
+              TARGET_TIMEOUT_CTRL. If the timeout is reached (or is configured to zero/disabled), the
+              pending write byte is NACK'd immediately.
+            - While stretching, writes that set TARGET_ACK_CTRL.NACK cause the pending write byte to
+              be NACK'd immediately.
+            - While stretching, the pending write byte is visible in ACQ_FIFO_NEXT_DATA.
+            - While stretching, writing a positive integer to the Auto ACK Counter causes the transfer
+              to continue, ACK'ing the pending data byte.
+
+            Stimulus
+            - Configure DUT/Agent to Target/Controller mode respectively, enable ACK control mode by
+              setting CTRL.ACK_CTRL_EN to 1.
+            - Generate write transfers, and allow the ACK Counter to reach 0 / timeout on stretch.
+            - While stretching, sometimes write to the TARGET_ACK_CTRL.NACK bit to NACK immediately.
+            - While stretching, sometimes write a new value to the ACK Counter to continue ACK'ing
+              automatically.
+
+            Checks
+            - Check the ACK Counter resets to 0 at the start of each transfer.
+            - Check that NACKs are generated when controlled to do so, or stretch-timeouts expire.
+            - When stretching, read back ACQ_FIFO_NEXT_DATA to check it matches the previously-written
+              data byte.
+            '''
+      stage: V2
+      tests: []
+    }
+    {
+      name: target_mode_bus_timeout
+      desc: '''
+            Test the bus_timeout feature (TIMEOUT_CTRL.MODE.BUS_TIMEOUT)
+
+            Stimulus
+            - Configure DUT/Agent as Target/Controller respectively.
+            - Configure TIMEOUT_CTRL with TIMEOUT_CTRL.MODE == BUS_TIMEOUT.
+            - Start a normal transaction using the agent, but halt midway-through and
+              hold SCL low.
+
+            Checking
+            - Both TARGET_EVENTS.BUS_TIMEOUT and CONTROLLER_EVENTS.BUS_TIMEOUT fields should be
+              set upon timeout.
```

Comment:
Note that these two have independent conditions:

`TARGET_EVENTS.BUS_TIMEOUT` only occurs if the bus times out during an active read transfer from our target.
`CONTROLLER_EVENTS.BUS_TIMEOUT` only occurs if the bus times out while our controller is actively engaged in a transaction.

If the DUT is configured as a Target only or if the agent is driving the transaction, then `CONTROLLER_EVENTS.BUS_TIMEOUT` will not be set.